fearless/Modules/Staking/Analytics/AnalyticsValidators/ViewModel/AnalyticsValidatorsViewModelFactory.swift

```diff
@@ -0,0 +1,287 @@
+                chartCenterText: chartCenterText,
+                listTitle: listTitle,
+                validators: validatorsWhoOwnedStake + validatorsWhoDontOwnStake,
+                selectedPage: page
+            )
+        }
+    }
+
+    private func activityProgressDescription(percents: Double, erasCount: Int) -> String {
+        let percentsString = percentFormatter.string(from: percents as NSNumber) ?? ""
+        return percentsString + " (\(erasCount) eras)"
+    }
+
+    private func findValidatorsWhoDontOwnStake(
+        page: AnalyticsValidatorsPage,
+        nomination: Nomination,
+        distinctValidators: Set<String>,
+        identitiesByAddress: [AccountAddress: AccountIdentity]?,
+        locale: Locale
+    ) -> [AnalyticsValidatorItemViewModel] {
+        let addressFactory = SS58AddressFactory()
+        let progressText: String = {
+            switch page {
+            case .activity:
+                return activityProgressDescription(percents: 0, erasCount: 0)
+            case .rewards:
+                return balanceViewModelFactory.amountFromValue(0).value(for: locale)
+            }
+        }()
+        return nomination
+            .targets
+            .compactMap { validatorId in
+                let validatorAddress = try? addressFactory.addressFromAccountId(
+                    data: validatorId,
+                    type: self.chain.addressType
+                )
+                guard
+                    let address = validatorAddress,
+                    !distinctValidators.contains(address)
+                else { return nil }
+
+                let icon = try? self.iconGenerator.generateFromAddress(address)
+                let validatorName = (identitiesByAddress?[address]?.displayName) ?? address
+                return AnalyticsValidatorItemViewModel(
+                    icon: icon,
+                    validatorName: validatorName,
+                    amount: 0,
+                    progressPercents: 0,
+                    progressText: progressText,
+                    validatorAddress: address
+                )
+            }
+    }
+
+    private func determineListTitle(page: AnalyticsValidatorsPage, locale: Locale) -> String {
+        switch page {
+        case .activity:
+            return R.string.localizable.stakingAnalyticsStakeAllocation(preferredLanguages: locale.rLanguages)
+        case .rewards:
+            return R.string.localizable.stakingRewardsTitle(preferredLanguages: locale.rLanguages)
+        }
+    }
+
+    private func createChartCenterText(
+        page: AnalyticsValidatorsPage,
+        validators: [AnalyticsValidatorItemViewModel],
+        totalEras: Int,
+        locale: Locale
+    ) -> NSAttributedString {
+        switch page {
+        case .activity:
+            let maxDistinctErasCount = validators.map(\.amount).max() ?? 0
+            let activeStakingErasPercents = Double(maxDistinctErasCount) / Double(totalEras)
```

Comment:
As agreed we need to display here total number of eras when a nominator had active staking